app/team.go

```diff
@@ -196,7 +207,18 @@ func (a *App) createOnboardingLinkedBoard(c request.CTX, teamId string) (*fb_mod
 	const preferenceName = "linked_board_created"
 	userId := c.Session().UserId
 
-	boardService := a.Srv().services[product.BoardsKey].(product.BoardsService)
+	boardServiceItf, ok := a.Srv().services[product.BoardsKey]
+	if !ok {
+		// here make sure to return error when necessary, othewise just log the errors
+		return nil, model.NewAppError("CreateBoard", "not product key found", nil, "", http.StatusBadRequest)
+	}
+	boardService, typeOk := boardServiceItf.(product.BoardsService)
+
+	if !typeOk {
```

Comment:
swapping where the empty line is to have the `if` close the what triggered it
```suggestion
	}

	boardService, typeOk := boardServiceItf.(product.BoardsService)
	if !typeOk {
```